### PR TITLE
Add federatedlearning in globalmanager

### PR DESCRIPTION
**Pull request by TymonXie**

### CODE REVIEW

File: build/crds/neptune/federatedlearningjob_v1alpha1.yaml [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+kind: CustomResourceDefinition
+metadata:
+  name: federatedlearningjobs.neptune.io
+spec:
+  group: neptune.io
+  names:
+    kind: FederatedLearningJob
+    plural: federatedlearningjobs
+    shortNames:
+      - fl
```
Comment (llhuii) [inline build/crds/neptune/federatedlearningjob_v1alpha1.yaml:11]: flj is enough

File: pkg/apis/neptune/v1alpha1/federatedlearningjob_types.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+// FLJobCondition describes current state of a job.
+type FLJobCondition struct {
+	// Type of job condition, Complete or Failed.
+	Type FLJobConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status v1.ConditionStatus `json:"status"`
+	// Last time the condition was checked.
+	// +optional
+	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty"`
```
Comment (llhuii) [inline pkg/apis/neptune/v1alpha1/federatedlearningjob_types.go:122]: rename to LastHeartbeatTime pls see https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties

File: build/crd-samples/neptune/federatedlearningjob_v1alpha.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        scriptBootFile: "train.py"
+        frameworkType: "tensorflow"
+        frameworkVersion: "2.3"
+        parameters:
+          - key: "batch_size"
+            value: "32"
+          - key: "learning_rate"
+            value: "0.001"
+          - key: "epochs"
+            value: "1"
```
Comment (llhuii) [inline build/crd-samples/neptune/federatedlearningjob_v1alpha.yaml:48]: pls add empty newline in end of file.

File: build/crds/neptune/federatedlearningjob_v1alpha1.yaml (hunk 1 above)
Comment (TymonXie) [reply build/crds/neptune/federatedlearningjob_v1alpha1.yaml:11]: i think it can be remained whitch seems more intuitive

Comment (llhuii) [reply build/crds/neptune/federatedlearningjob_v1alpha1.yaml:11]: then use `fl`, one short name is enough

File: pkg/apis/neptune/v1alpha1/federatedlearningjob_types.go (hunk 2 above)
Comment (TymonXie) [reply pkg/apis/neptune/v1alpha1/federatedlearningjob_types.go:122]: ok，i'll update this uniformly in next version

Comment (llhuii) [reply pkg/apis/neptune/v1alpha1/federatedlearningjob_types.go:122]: remaind that crd yaml is also needed to be updated if you decide to update.

**Review by llhuii — APPROVED** [review]

**Pull request merged**